### PR TITLE
[DEV-1297] support tool v2

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations
  - Login/logout
  - Registration
  - Update user data
  - Create/delete payment routes

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Comment by Yannick1712** [comment]
buyCryptoTxId is not unique - no sense to use it or? 🤔

**Review by davidleomay — COMMENTED** [review]
Restructure the query DTO to:
```ts
{
  table: 'string',
  key: 'string',
  value: any
}
```

File: src/subdomains/generic/gs/gs.service.ts [hunk 1]
```diff
@@ -97,18 +114,41 @@ export class GsService {
...
+      case SupportTable.USER:
+        return await this.userService.getUserByKey(query.key, query.value).then((user) => user?.userData);
+      case SupportTable.BUY:
+        return await this.buyService.getBuyByKey(query.key, query.value).then((buy) => buy?.user.userData);
+      case SupportTable.SELL:
+        return await this.sellService.getSellByKey(query.key, query.value).then((sell) => sell?.user.userData);
+      case SupportTable.BUY_CRYPTO:
+        return await this.buyCryptoService
+          .getBuyCryptoByKey(query.key, query.value)
+          .then((buyCrypto) => (buyCrypto?.buy ? buyCrypto?.buy.user.userData : buyCrypto?.cryptoRoute?.user.userData));
```
Comment (davidleomay) [inline src/subdomains/generic/gs/gs.service.ts:129]: Can we use `buyCrypto?.user.userData` here?

File: src/mix/models/crypto-input/crypto-input.entity.ts [hunk 2]
```diff
@@ -58,7 +58,7 @@ export class CryptoInput extends IEntity {
   @ManyToOne(() => Asset, { nullable: false, eager: true })
   asset: Asset;
 
-  @Column({ length: 256, nullable: false })
+  @Column({ length: 256, nullable: true })
```
[the remark this answers is not in the document]
Comment (Yannick1712) [reply src/mix/models/crypto-input/crypto-input.entity.ts:61]: Ups 😅

**Review by davidleomay — COMMENTED** [review]
Fix linter errors

**Review by davidleomay — APPROVED** [review]

**Pull request merged**